components/scream/src/share/io/scorpio_input.hpp

```diff
@@ -93,6 +93,7 @@ class AtmosphereInput
   //               variables from the input filed will be read into.
   //               Fields can be padded/strided.
   // It calls init(field_mgr) at the end.
+  // TODO: is comm superfluous, considering we can get it from the grid in the field_mgr?
```

Comment:
This is a good question! I think the answer is the same as the answer to "is it invalid for this comm to be different from the one used by the grid in the field_mgr?"

Reply:
True. I can see how this could be potentially buggy, so maybe I should remove the comm from the inputs. And if one uses that basic ctor (comm+params) and calls `init(field_mgr,grids_mgr)` later, we should check that the stored comm matches the one in the field mgr (and grids mgr)